### ISSUE TITLE
Implement bending resistance

### ISSUE THREAD

**Issue by FinFetChannel** [mattpalermo/chain-fountain-simulator#7]
Is it possible to introduce a bending restrictor? could be a simple  minimum angle between two links that are connected to the same element, or a full fledged transversal spring model, but that may be overkill. I think this way it could be more realistic when considering some types of chains

**Comment by FinFetChannel** [comment]
well, now looking at how the chai is stored on the beaker this may be problematic, in the real world the chain bends in the third dimension which is not possible here

**Comment by mattpalermo** [comment]
Hi FinFET. I think it would be possible using a spring like force. A simple bending restriction wouldn't work well with the integration method used at the moment. The initial placement of the beads might have to be changed? The 2d is ok because the chain is free to overlap and pass through itself.

**Comment by FinFetChannel** [comment]
Hello Matt! Yes, I was thinking about this initial placement, the beads would be spring loaded with an 180° bend, the chain would probably "explode".

**Comment by mattpalermo** [comment]
Maybe. Depends how strong the restoring force is set to. Have you got an idea of how the force on each bead should be calculated based on the position of it's two neihbouring beads? At the moment it is modelled as if each bead is connected by springs.

**Comment by FinFetChannel** [comment]
I was thinking about something like a torsion spring, maybe linear with the angle, but chains irl have some low angles where they can bend with  almost no resistance and after that they sort of lock. This could allow for an initial placement with a bulge on either side without it springing, all elements below this critical angle. I fear that this implementation could slow down the simulation due to the increase in complexity.

**Comment by mattpalermo** [comment]
True. The force can be any function of position, but it cannot be a function of velocity. So it doesn't have to be linear. And yea it will slow it down. It would be best to have it off by default because, despite Biggins and Warners explaination, a minimum radius of curvature seems to be unnessesary to forming a chain fountain.

**Comment by FinFetChannel** [comment]
I understand, I think you can close this issue if you want. I was intrigued by this phenomena and wanted to understand better the mechanism behind the formation of the raised loop. From the footage available online and experimentation with your simulation my main hypothesis is that the waves that form when the chain is pulled have a major role. I found at least one paper which has a similar conclusion https://arxiv.org/ftp/arxiv/papers/1810/1810.13008.pdf
